framework/src/problems/EigenProblem.C

```diff
@@ -403,18 +388,21 @@ EigenProblem::initEigenvector(const Real initial_value)
     if (!n_dofs)
       break;
 
-    const auto & var = getVariable(0, vn);
+    MooseVariableBase * var = nullptr;
+    if (hasScalarVariable(vn))
+      var = &getScalarVariable(0, vn);
+    else
+      var = &getVariable(0, vn);
     // We set values for only eigen variables
-    if (var.eigen())
+    if (var->eigen())
     {
       std::set<dof_id_type> var_indices;
-      for (unsigned int vc = 0; vc < var.count(); ++vc)
+      for (unsigned int vc = 0; vc < var->count(); ++vc)
       {
         var_indices.clear();
-        _nl_eigen->system().local_dof_indices(var.number() + vc, var_indices);
+        _nl_eigen->system().local_dof_indices(var->number() + vc, var_indices);
         for (const auto & dof : var_indices)
-          // Davidson by n_dofs is a simple way to nondimensionalize eigen vectors
-          _nl_eigen->solution().set(dof, initial_value / n_dofs);
```

Comment:
It is interesting here. Why you removed `n_dofs`? Because you always do an expensive `prescale` regardless?

Reply:
@YaqiWang

Reply:
Yes. I feel it is hard for users to understand we did this division as well.

Reply:
Overall, only the shape of the solution matters to the solver. I feel we should make this as simple as possible, i.e. just set to the value users asked for in the input file.

Reply:
It was necessary before,  but we do not need this anymore because we have an expensive `prescale` function now